tools/sortimports/main.go

```diff
@@ -0,0 +1,257 @@
+/*
+ * Copyright 2025 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package main
+
+import (
+	"bytes"
+	"flag"
+	"fmt"
+	"go/ast"
+	"go/format"
+	"go/parser"
+	"go/printer"
+	"go/token"
+	"io/fs"
+	"log"
+	"os"
+	"path/filepath"
+	"sort"
+	"strings"
+
+	"k8s.io/apimachinery/pkg/util/sets"
+)
+
+const (
+	// the below values taken from src/go/format/format.go
+	tabWidth                = 8
+	printerNormalizeNumbers = 1 << 30
+	printerMode             = printer.UseSpaces | printer.TabIndent | printerNormalizeNumbers
+)
+
+var config = printer.Config{Mode: printerMode, Tabwidth: tabWidth}
+
+type Args struct {
+	Paths       []string
+	ExcludeDirs sets.Set[string]
+	Verbose     bool
+}
+
+// Sorts imports based on:
+// 1. Standard libraries
+// 2. "k8s.io" imports
+// 3. "sigs.k8s.io" imports
+// 4. Other third-party libraries
+// 5. "github.com/openshift" imports
```

Comment:
it is probably not common to use packages from other openshift projects but maybe we can add another filter specific for the local nrop imports and place them at the bottom:
`github.com/openshift-kni/numaresources-operator`